Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/plan_create_task_list/PlanTaskListFragment.java

```diff
@@ -2,43 +2,74 @@
 
 import android.app.Fragment;
 import android.app.FragmentTransaction;
+import android.content.Intent;
 import android.databinding.DataBindingUtil;
 import android.os.Bundle;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
 import javax.inject.Inject;
 
+import database.entities.PlanTaskTemplate;
 import database.entities.PlanTemplate;
 import database.repository.PlanTemplateRepository;
 import pg.autyzm.friendly_plans.ActivityProperties;
 import pg.autyzm.friendly_plans.App;
 import pg.autyzm.friendly_plans.R;
 import pg.autyzm.friendly_plans.databinding.FragmentPlanTaskListBinding;
+import pg.autyzm.friendly_plans.item_touch_helper.SimpleItemTouchHelperCallback;
+import pg.autyzm.friendly_plans.manager_app.view.main_screen.MainActivity;
 import pg.autyzm.friendly_plans.manager_app.view.plan_create_add_tasks.AddTasksToPlanFragment;
-import pg.autyzm.friendly_plans.manager_app.view.task_list.TaskRecyclerViewAdapter;
+import pg.autyzm.friendly_plans.notifications.ToastUserNotifier;
 
 public class PlanTaskListFragment extends Fragment implements PlanTaskListEvents {
 
-    private TaskRecyclerViewAdapter taskListAdapter;
+    private PlanCreateTaskListRecyclerViewAdapter taskListAdapter;
     private Long planId;
 
     @Inject
     PlanTemplateRepository planTemplateRepository;
+    @Inject
+    ToastUserNotifier toastUserNotifier;
+
+    PlanCreateTaskListRecyclerViewAdapter.TaskItemClickListener taskItemClickListener =
```

Comment:
I wonder if that code could be extracted to separate  class implementing this interface, only create instance here ( dependencies could be pass in constructor) or are there any Android limitations?